### PR TITLE
lots of changes...

### PR DESCRIPTION
**Pull request by mingkaili**
My apologies for quite a big PR. Normally, we would want to limit the size of PRs for the sake of reviewers, but bc of the time constraint we are in, I am combining 3 issues into 1 single issue. Here is a list of things I have done:
1. login modal
2. login submit function
3. fetch call to get the token
4. load user specific challenges if the user is logged in
5. signup modal
6. signup submit function
7. fetch call to signup user and auto login the user
8. getChallenge function to load from server
9. Recipe display merged from https://github.com/cse112-group10/cse112-sp22-group10/pull/45
10. Complete challenge functionality

### CODE REVIEW

File: source/frontend/assets/components/RecipeDisplay.js [hunk 1]
```diff
@@ -496,25 +500,23 @@ class RecipeDisplay extends HTMLElement {
...
-        imgPreview.style.display = '';
-        const uploadImg = this.shadowRoot.getElementById('imgUpload');
-        uploadImg.style.display = '';
-        const submitBtn = this.shadowRoot.getElementById('submitButton');
-        submitBtn.style.display = '';
-        this.GetImgurImage();
-        this.json.reactions = 'assets/images/placeholder.png';
-        this.SubmitReaction();
-      }
+      await database.completeRecipe(data);
```
Comment (joecai6) [inline source/frontend/assets/components/RecipeDisplay.js:505]: I will make another PR to remove this and add the frontend for displaying a completed recipe.

Comment (mingkaili) [reply source/frontend/assets/components/RecipeDisplay.js:505]: I am not sure what you mean by that. This is the function that gets called when the complete button is pressed, it has nothing to do with displaying a completed reecipe or removing it won't help with that in anyway.

Comment (joecai6) [reply source/frontend/assets/components/RecipeDisplay.js:505]: I selected the wrong lines, i meant the lines below where it shows the input for an uploading image.

Comment (mingkaili) [reply source/frontend/assets/components/RecipeDisplay.js:505]: ok

**Pull request merged**